bigbluebutton-html5/imports/ui/components/chat/message-form/typing-indicator/container.jsx

```diff
@@ -19,31 +20,39 @@ class TypingIndicatorContainer extends PureComponent {
 
 export default withTracker(() => {
   const idChatOpen = Session.get('idChatOpen');
+  const meeting = Meetings.findOne({ meetingId: Auth.meetingID }, {
+    fields: {
+      'lockSettingsProps.hideUserList': 1,
+    },
+  });
 
-  let selector = {
+  const selector = {
     meetingId: Auth.meetingID,
     isTypingTo: PUBLIC_CHAT_KEY,
   };
 
   if (idChatOpen !== PUBLIC_CHAT_KEY) {
-    selector = {
-      meetingId: Auth.meetingID,
-      isTypingTo: Auth.userID,
-      userId: idChatOpen,
-    };
+    selector.isTypingTo = Auth.userID;
+    selector.userId = idChatOpen;
   }
 
-  const typingUsers = UsersTyping.find(selector).fetch();
-
   const currentUser = Users.findOne({
     meetingId: Auth.meetingID,
     userId: Auth.userID,
   }, {
     fields: {
       userId: 1,
+      role: 1,
     },
   });
 
+  if (meeting.lockSettingsProps.hideUserList && currentUser.role === 'VIEWER') {
+    selector.role = { $ne: 'VIEWER' };
```

Comment:
Can you change this to pull the 'VIEWER' string from settings.yml please. There's a `role_viewer` property that you can use.

Reply:
Also the literal that's on ln 49 also.